### PR TITLE
ZER-185 Change text in drop down lists in data input block

### PR DESCRIPTION
**Pull request by popovycj**
dev
## JIRA

* [ZER-185](https://jira.softserve.academy/browse/ZER-185)

## Code reviewers

- [ ] @loqimean 

## Summary of issue

As a user, I want to see numbers with words in drop-down lists so that the choice will be easier for understanding.

Acceptance criteria:

GIVEN the user wants to choose the child’s age on the calculator page
WHEN he/she is clicking on ‘years’ and ‘months’
THEN ‘0 years; 1 year; 2 years’ is displayed for ‘years’
AND ‘1 month; 0, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 months’ is displayed for ‘months’
See the exact texts in the file: https://docs.google.com/document/d/1ezwQR3gtY06rKcotLiFJE_XJBqnw5I1RuzTHiywW4eU/edit

English:
image

Ukrainian:
image

## Summary of change

English:
Added words 'year' and 'years' for the corresponding values in years dropdown. Similar for months dropdown

image

image

Ukrainian:
Added words 'років', 'рік', 'роки' for the corresponding values in years dropdown. Similar for months dropdown

image

image

Additional changes:
Set attribute 'hidden' for values above '0 years' or '0 months' values

## Testing approach

ToDo

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: config/locales/uk.yml [hunk 1]
```diff
@@ -12,6 +12,14 @@ uk:
         budgetary: "бюджетна"
         medium: "середня"
         premium: "преміум"
+        years:
+          zero: "0 років"
+          one: "1 рік"
+          few: "%{count} роки"
```
Comment (loqimean) [inline config/locales/uk.yml:18]: `10 роки` ?)

Comment (loqimean) [reply config/locales/uk.yml:18]: add to project [this](https://github.com/svenfuchs/rails-i18n/blob/master/rails/locale/uk.yml) file and use

File: spec/lib/inflector_extensions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe '#pluralize' do
```
Comment (loqimean) [inline spec/lib/inflector_extensions_spec.rb:3]: не то, юзай клас, який тестуєш

File: spec/lib/inflector_extensions_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe '#pluralize' do
+  context 'months' do
```
Comment (loqimean) [inline spec/lib/inflector_extensions_spec.rb:4]:
```suggestion
  context 'with months' do

```

Comment (loqimean) [reply spec/lib/inflector_extensions_spec.rb:4]: and so on below

File: lib/inflector_extensions.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    else
+      ActiveSupport::Inflector.pluralize(self, count, locale)
+    end
+  end
+end
+
+module InflectorExtensions
+  def pluralize(word, count = nil, locale = :en)
+    apply_inflections(word, inflections(locale).plurals, locale, count)
+  end
```
Comment (loqimean) [inline lib/inflector_extensions.rb:17]: add create specs also just for class `Inflections` and do something like `Inflections.pluralize('something', 2)   #=>  "somethings"`

File: spec/lib/inflector_extensions_spec.rb [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    end
+  end
+end
+
+RSpec.describe ActiveSupport::Inflector do
+  context 'with #pluralize' do
+    it 'pluralizes properly' do
+      expect(ActiveSupport::Inflector.pluralize('місяць', 2, :uk)).to eq('місяці')
+      expect(ActiveSupport::Inflector.pluralize('рік', 0, :uk)).to eq('років')
+      expect(ActiveSupport::Inflector.pluralize('dog', locale = :en)).to eq('dogs')
```
Comment (loqimean) [inline spec/lib/inflector_extensions_spec.rb:91]: не треба тут це `locale = something`

File: lib/inflector_extensions.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  def apply_inflections(word, rules, locale = :en, count = nil)
+    result = word.to_s.dup
+
+    return result if word.empty? \
+                       || inflections(locale).uncountables.uncountable?(result)
+
+    rules.each do |rule, replacement, range|
+      break if (range.nil? || range.include?(count)) \
```
Comment (loqimean) [inline lib/inflector_extensions.rb:26]: нашо тут цей бекслеш?

Comment (VasylenchukMischa) [reply lib/inflector_extensions.rb:26]: ага, краще перепиши умову на вигляд if ... end

**Review by VasylenchukMischa — APPROVED** [review]

File: lib/inflector_extensions.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class String
+  def pluralize(count = nil, locale = :en)
+    locale = count if count.is_a?(Symbol)
```
Comment (loqimean) [inline lib/inflector_extensions.rb:5]:
```suggestion
    locale = count if count.is_a?(Symbol)

```

Comment (loqimean) [reply lib/inflector_extensions.rb:5]: чому у тебе `count` може бути символом?

File: lib/inflector_extensions.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class String
+  def pluralize(count = nil, locale = :en)
+    locale = count if count.is_a?(Symbol)
+    if count == 1
+      dup
```
Comment (loqimean) [inline lib/inflector_extensions.rb:7]: це для чого?

File: lib/inflector_extensions.rb (hunk 7 above)
Comment (loqimean) [reply lib/inflector_extensions.rb:5]: ну та зроби це неймед параметрами, нашо зайва робота, ну чи каунт тік зроби неймид

File: lib/inflector_extensions.rb (hunk 8 above)
Comment (loqimean) [reply lib/inflector_extensions.rb:8]: нехай буде

File: lib/inflector_extensions.rb (hunk 7 above)
Comment (popovycj) [reply lib/inflector_extensions.rb]: changed

**Comment by VasylenchukMischa** [comment]
https://stackoverflow.com/questions/6166064/i18n-pluralization

**Comment by VasylenchukMischa** [comment]
таким чином пробував?

**Comment by popovycj** [comment]
> таким чином пробував?

Та, можеш глянути перші коміти. А так, то рішили з Іваном піти цікавішим шляхом)

**Review by VasylenchukMischa — APPROVED** [review]

**Pull request merged**